From b9bc3afc9eb85a7fb9a825eec91a28db37dc2474 Mon Sep 17 00:00:00 2001
From: Harsh J <harsh@cloudera.com>
Date: Fri, 16 Oct 2015 17:12:35 +0200
Subject: [PATCH 1934/2848] HDFS-9257. improve error message for "Absolute
 path required" in INode.java to contain the
 rejected path. Contributed by Marcell Szabo

(cherry picked from commit 52ac73f344e822e41457582f82abb4f35eba9dec)

Change-Id: I3b14ce0decb775884c5afbe4f051978a20071efd
---
 .../apache/hadoop/hdfs/server/namenode/INode.java  |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INode.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INode.java
index 40f1755..17991e0 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INode.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INode.java
@@ -791,7 +791,8 @@ public final INode setAccessTime(long accessTime, int latestSnapshotId) {
 
   private static void checkAbsolutePath(final String path) {
     if (path == null || !path.startsWith(Path.SEPARATOR)) {
-      throw new AssertionError("Absolute path required");
+      throw new AssertionError("Absolute path required, but got '"
+          + path + "'");
     }
   }
 
-- 
1.7.9.5

